types/save-csv/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for save-csv 4.1
+// Project: https://github.com/silverwind/save-csv
+// Definitions by: Frank Brullo <https://github.com/FrankBrullo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.1
+
+export interface SaveCsvProps {
+    filename: string;
+    sep?: string;
+    eol?: string;
+    quote?: string;
+    bom?: boolean;
+    mime?: string;
+    formatter?: (value: string | JSON) => void;
+}
+
+export default function saveCsv(array: ReadonlyArray<any>, options?: SaveCsvProps): void;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
Thank you @DanielRosenwasser . The documentation was very helpful.
I tried before like you said but it didn't work (probably because I didn't know the `esModuleInterop` switch.
At the end of the day, I used `export default` because it was working out-of-the-box with a default project created with create-react.
If I apply your changes, how can I make sure that anybody else will know that they have to use such that switch once they download the type definition?

Reply:
I think that's up to the TypeScript project - we should provide a better error message in those cases. I just opened up https://github.com/Microsoft/TypeScript/issues/30837

Reply:
@DanielRosenwasser I am not really happy to implement a solution that is syntactically perfect but requires manual fiddling from the user once they install the type.
I would rather prefer a less perfect solution but which works in any case.

Anyway if this is how you want it for the sake of moving this forward I will do it.